golang/pre-publish/action.yml

```diff
@@ -16,8 +16,30 @@ runs:
         version: ${{ inputs.version }}
         version_bump_script: "go run ${{ github.action_path }}/bump-version.go"
         commit_template: "BUMP v${VERSION}"
-        push_commit: ${{ inputs.push_changes }}
+        # Never push commit, we still need to merge up if a push is requested
+        push_commit: false
     - uses: mongodb-labs/drivers-github-tools/tag-version@v2
       with:
         version: v${{ inputs.version }}
-        push_tag: ${{ inputs.push_changes }}
\ No newline at end of file
+        push_tag: ${{ inputs.push_changes }}
+    - name: "Determine branch to merge up to"
+      if: ${{ inputs.push_changes }}
+      id: get-next-branch
+      uses: alcaeus/automatic-merge-up-action/get-next-branch@b253eae7dfccc0b637edf71121608375e07a7678 #main
+      with:
+        ref: ${{ github.ref_name }}
+        branchNamePattern: 'release/<major>.<minor>'
+        devBranchNamePattern: 'v<major>'
+        fallbackBranch: 'master'
+    - name: "Manually merge up changes"
+      if: ${{ inputs.push_changes && steps.get-next-branch.outputs.hasNextBranch }}
+      shell: bash
+      run: |
+        git checkout ${NEXT_BRANCH}
+        git merge --strategy=ours ${RELEASE_BRANCH}
```

Comment:
Part of the release process bumps the patch version. Would this end up creating a PR to bump the version on a minor release > that the one being targeted by the action? For example if you are releasing 2.1.1 and a release/2.2 branch exists would a PR be merged up to release/2.2 that attempts to change the version to 2.1.1?

Reply:
No. Let's assume you are releasing 2.1.1. Pushing that commit would usually trigger a pull request to merge that change into release/2.2. You'd have to manually ignore the changes in that one and merge the commit as merged.

With this change, we're manually merging the bump commit into the release/2.2 branch, but by using the `ours` strategy, no changes will be applied. Since release/2.2 then receives a push, it will trigger a merge-up of the commit (with no changes) to the next branch.

The main difference here is that this takes care of the first merge-up PR that would require manual intervention to not change the version in any newer branches. From there, it's just a straightforward approval to enable auto-merge until the end of the chain.